libs/ui/src/components/NeoMessage/NeoMessage.vue

```diff
@@ -1,146 +1,137 @@
-      </section>
+      </div>
+
+      <div
+        v-if="showProgressBar"
+        class="w-full h-1 message-progress absolute left-0 bottom-0 transition-all ease-linear"
+        :style="{ width: `${percent}%` }" />
     </article>
   </transition>
 </template>
 
-<script lang="ts">
+<script lang="ts" setup>
+import { useElementHover } from '@vueuse/core'
 import NeoButton from '../NeoButton/NeoButton.vue'
 import NeoIcon from '../NeoIcon/NeoIcon.vue'
+import { NeoMessageVariant } from '../../types'
 
-export default {
-  name: 'NeoMessage',
-  components: {
-    NeoButton,
-    NeoIcon,
-  },
-  props: {
-    active: {
-      type: Boolean,
-      default: true,
-    },
-    title: String,
-    closable: {
-      type: Boolean,
-      default: true,
-    },
-    message: String,
-    type: String,
-    hasIcon: Boolean,
-    size: String,
-    icon: String,
-    iconPack: String,
-    iconSize: String,
-    autoClose: {
-      type: Boolean,
-      default: false,
-    },
-    duration: {
-      type: Number,
-      default: 2000,
-    },
-  },
-  data() {
-    return {
-      isActive: this.active,
-      remainingTime: this.duration / 1000, // in seconds
-    }
-  },
-  computed: {
-    computedIcon() {
-      if (this.icon) {
-        return this.icon
-      }
-      switch (this.type) {
-        case 'is-info':
-          return 'information'
-        case 'is-success':
-          return 'check-circle'
-        case 'is-warning':
-          return 'alert'
-        case 'is-danger':
-          return 'alert-circle'
-        default:
-          return null
-      }
-    },
-  },
-  watch: {
-    active(value) {
-      this.isActive = value
-    },
-    isActive(value) {
-      if (value) {
-        this.setAutoClose()
-      } else {
-        if (this.timer) {
-          clearTimeout(this.timer)
-        }
-      }
-    },
-  },
-  mounted() {
-    this.setAutoClose()
-  },
-  methods: {
-    /**
-     * Close the Message and emit events.
-     */
-    close() {
-      this.isActive = false
-      this.resetDurationProgress()
-      this.$emit('close')
-      this.$emit('update:active', false)
-    },
-    click() {
-      this.$emit('click')
-    },
-    /**
-     * Set timer to auto close message
-     */
-    setAutoClose() {
-      if (this.autoClose) {
-        this.timer = setTimeout(() => {
-          if (this.isActive) {
-            this.close()
-          }
-        }, this.duration)
-      }
-    },
-    resetDurationProgress() {
-      /**
-       * Wait until the component get closed and then reset
-       **/
-      setTimeout(() => {
-        this.remainingTime = this.duration / 1000
-      }, 100)
-    },
+const iconVariant: Record<NeoMessageVariant, string> = {
+  info: 'circle-info',
+  success: 'check-circle',
+  warning: 'circle-exclamation',
+  danger: 'circle-exclamation',
+}
+
+const emit = defineEmits(['close', 'update:active', 'click'])
+const props = withDefaults(
+  defineProps<{
+    title?: string
```

Comment:
same here

Reply:
since you can pass title as a slot , this can be nullable , default `''` ?

Reply:
yes, how about `''`

Reply:
updated